utils/database.go

```diff
@@ -0,0 +1,28 @@
+package utils
+
+import (
+	"fmt"
+	"log"
+	"os"
+
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+)
+
+var DB *gorm.DB
+var err error
```

Comment:
Defining exported global variables can be really handy in some situations. However, it comes with many downsides. Can you mention some of them? How would you approach this if you weren't allowed to use global variables?

Reply:
Some of the downsides using global variables are:

- Global variables have no access-control. We can access and change these variables in any part of the code, 
 thus it is difficult to debug.
- Global variables are not in harmony with Open-Close design SOLID principle 

Alternatively, I could have created struct and a repository interface in that will hold the the database dependency.

    type Database struct{
       Db *gorm.DB
    }
    
    func DatabaseInterface(db *gorm.DB) *Database {
       return &Database{Db: db}
    }

and use them in corresponding functions

Reply:
Great points made!